services/wallet/api.go

```diff
@@ -230,6 +237,20 @@ func (api *API) GetPendingTransactionsByChainIDs(ctx context.Context, chainIDs [
 	return rst, err
 }
 
+func (api *API) GetPendingTransactionsForIdentities(ctx context.Context, identities []transfer.TransactionIdentity) (result []*transfer.PendingTransaction, err error) {
```

Comment:
Why not mixing with the rest? Out of curiosity can we just have GetActivityEntries?

Reply:
I wanted to keep the separation of concerns at the current level. The `GetActivityEntries` should only provide the metadata based on filter criteria so that there is no need for merging the results on the Nim side.

Now I'm thinking even further with this and looking into `GetActivityEntries` serving all the needs for the activity summary. These details APIs should only be called when a user views the transaction details (with a sub-transactions list).

Does it make sense? Do you see some inconveniences or risks?

From a performance point of view seems optimal; most of the important details we have them because we need to take them in account for the filter when we query. Extending with more columns and joining other tables sounds like a small overhead.

Reply:
> So this one will be remove in the upcoming PRs?

Yes, or converted to extract only one entry. Or used like this for the case of one entry and multiple after analysing the requirements for pending transactions details view.

Reply:
> So this one will be remove in the upcoming PRs?

Yes, or converted to extract only one entry. Or used like this for the case of one entry and multiple after analysing the requirements for pending transactions details view.